### PR TITLE
Centralize booking payment and failure rules

### PR DESCRIPTION
**Pull request by stefan-burke**
## Summary

- replace the inline payment condition with one pure plan that resolves the effective price, checkout path, and provider-less balance
- define the recoverable attendee creation and update failures with shared Valibot schemas
- derive database result types and exhaustive failure messages from those schemas
- treat a capacity loss during atomic creation as the same `sold_out` outcome as the preflight check
- remove `encryption_error` from results, API responses, admin fallbacks, refund codes, copy, and tests
- use the request/setup-provided encryption keys directly, without presence checks or tests that corrupt the atomic setup state
- document the request key invariants in `AGENTS.md`
- mark both business-logic TODOs as shipped

## Checks

- `deno task precommit`
- affected attendee/PII test files (54 tests passed)
- `deno task mutation src/shared/booking/payment-plan.ts test/shared/booking/payment-plan.test.ts` (12/12 mutants killed)
- `deno task mutation src/shared/attendee-failures.ts test/shared/attendee-failures.test.ts` (4/4 mutants killed)
- `deno task mutation src/shared/db/attendees/pii.ts test/shared/db/attendees/pii.test.ts --harness` (22/22 mutants killed; 3 proven equivalents)

## Summary by CodeRabbit

* **New Features**
  * Added unified booking payment handling for checkout, free bookings, custom pricing, and unpaid balances.
  * Improved attendee failure messaging, including named sold-out listings and fallback messages.

* **Bug Fixes**
  * Standardized booking and servicing responses for capacity-related failures.
  * Improved sold-out detection and checkout pricing at boundary values.

* **Tests**
  * Added coverage for payment scenarios and attendee failure formatting.
  * Updated existing tests for revised booking and attendee error behavior.

* **Documentation**
  * Updated tracking documentation to reflect completed booking payment and failure-handling work.

### CODE REVIEW

File: src/shared/booking/payment-needed.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/** Whether a booking must open a payment checkout for its configured price. */
+export const bookingNeedsPayment = (
+  paymentsEnabled: boolean,
+  unitPrice: number,
+  customUnitPrice?: number,
+): boolean =>
+  paymentsEnabled &&
+  (unitPrice > 0 || (customUnitPrice !== undefined && customUnitPrice > 0));
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/shared/booking/payment-needed.ts]: Resolved in f388817a. The old predicate was replaced by `planBookingPayment`, which derives the effective price with `customUnitPrice ?? unitPrice` before selecting checkout. `test/shared/booking/payment-plan.test.ts` now covers base price 100 with an explicit custom price of 0 and expects a direct zero-balance plan; the targeted mutation run also kills `??` to `||`.

File: src/shared/booking/form.ts [hunk 2]
```diff
@@ -28,4 +28,4 @@ export const bookingError = {
 };
 
 /** Format error message for failed attendee creation. */
-export const formatAtomicError = capacityErrorFormatter(bookingError);
+export const formatAtomicError = attendeeFailureFormatter(bookingError);
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/shared/booking/form.ts]: Resolved in 05d0d17a. `attendee-failures.ts` now exports the named `AttendeeFailureFormatter` contract, and `formatAtomicError` is explicitly annotated with it while keeping the same initializer. This reuses the shared callable contract rather than repeating the signature; `deno task precommit` passes.

File: src/shared/db/attendees/servicing.ts [hunk 3]
```diff
@@ -270,12 +269,9 @@ export const createServicingEvent = async (
...
   if (!createResult.success) {
-    const names =
-      createResult.reason === "capacity_exceeded"
-        ? await joinedListingNames(
-            unique(input.bookings.map((booking) => booking.listingId)),
-          )
-        : "";
+    const names = await joinedListingNames(
+      unique(input.bookings.map((booking) => booking.listingId)),
+    );
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/shared/db/attendees/servicing.ts:274]: Not actionable in the current model. `createAttendeeAtomic` returns `CreateAttendeeResult`, whose failure reason is `AttendeeCreationFailureReason`, the literal `capacity_exceeded`; `no_lines` exists only on `UpdateAttendeeAtomicResult`. The servicing input asserter also rejects empty bookings before this call, so this lookup is reached only for the capacity failure whose message needs listing names. Adding a `no_lines` branch here would check an impossible state and contradict the request/setup invariant policy added in 02d924c1.

**Comment by stefan-burke** [comment]
@coderabbitai The outside-diff `formatPostPaymentError` finding is not actionable. The formatter is called with `result.reason`, where `result` is the return from `attendeesApi.createBookingAtomic`; after the preceding `result === "sold-out"` branch, its only failure shape is `CreateAttendeeResult` with the literal `capacity_exceeded` reason. `unexpected_error` belongs to the broader `HonourResult` return contract but is returned earlier from the preparation `catch` and is never passed to this formatter. The checked types preserve that narrowing, so adding payment-wide dispatch here would model an impossible path.
